File: Dataset/cpp/highmiddlelow.cpp

//  highmiddlelow.cpp (c) 2002 Kari Laitinen

#include  <iostream.h>

class  Member_class
{
public:
   Member_class()
   {
      cout << "\n Member_class constructor was called. " ;
   }

   ~Member_class()
   {
      cout << "\n Member_class destructor was called. " ;
   }
} ;

class  High_class 
{
protected:
   Member_class  some_data_member ;
   Member_class  another_data_member ;

public:
   High_class()
   {
      cout << "\n The constructor of High_class was called." ;
   }

   ~High_class()
   {
      cout << "\n The destructor of High_class was called." ;
   }
} ;

class  Middle_class  :  public  High_class
{
public:
   Middle_class()
   {
      cout << "\n The constructor of Middle_class was called." ;
   }

   ~Middle_class()
   {
      cout << "\n The destructor of Middle_class was called." ;
   }
} ;

class  Low_class  :  public  Middle_class
{
protected:
   Member_class  data_member_in_low_class ;

public:
   Low_class()
   {
      cout << "\n The constructor of Low_class was called." ;
   }
   ~Low_class()
   {
      cout << "\n The destructor of Low_class was called." ;
   }
} ;

int main()
{
   Low_class  low_class_object ;
}

File: Dataset/cs/Highmiddlelow.cs

//  Highmiddlelow.cs (c) 2003 Kari Laitinen

using System ;

class  MemberClass
{
   public MemberClass( string  object_identifier )
   {
      Console.Write( "\n MemberClass object \"" 
                    +  object_identifier  + "\" was created." ) ;
   }
}

class  HighClass 
{
   MemberClass  some_data_member ;
   MemberClass  another_data_member 
                    =  new  MemberClass( "another_data_member" ) ;

   public HighClass()
   {
      Console.Write( "\n The constructor of HighClass started.") ;

      some_data_member  =  new  MemberClass( "some_data_member" ) ;

      Console.Write( "\n The constructor of HighClass ended.") ;
   }
}

class  MiddleClass  :  HighClass
{
   public MiddleClass()
   {
      Console.Write( "\n The constructor of MiddleClass started.");
      Console.Write( "\n The constructor of MiddleClass ended.");
   }

}

class  LowClass  :  MiddleClass
{
   MemberClass data_member_in_low_class  =
                   new MemberClass( "data_member_in_low_class" );

   public LowClass()
   {
      Console.Write( "\n The constructor of LowClass started." ) ;
      Console.Write( "\n The constructor of LowClass ended." ) ;
   }
} 

class Highmiddlelow
{
   static void Main()
   {
      LowClass  low_class_object  =  new  LowClass() ;
   }
}



